### PR TITLE
Re-structuring `client` & Renew UI

### PR DESCRIPTION
**Pull request by Prev**
Make components re-usable and separate their roles.

### 1. Project tree is modified.

_Before:_
- /containers
- /actions
- /reducers
- /forms

_After:_
- /components -> Re-usable components. (No side-effect)
- /containers -> Side-effect, Ajax, correspond to router DOM. (Maybe called like `modules`)
    - actions.js
    - reducer.js
    - index.js (Container component)
    - component.js (Optional. Presentational component of this model/container)
    - forms, css...

### 2. `User` and `Report` features are removed.

Features for managing `User` would not be included to `client package`.
Make `admin` or `dashboard` package in soon.

There is no specific requirements for `Report` now, so it is confusing codes.
This feature would be implemented later.

### 3. `NodeList` is renewed to `NodeView`.

All variables named `parent` and `nodes` are changed to `node` and `children`.  
From now, this module's role is not to **LIST** nodes, but show node's content and it's children.

There are two types of node to _view_, `Forum` and `Topic`.
`Reply` is **weak**-type node, can not be stand-alone, so there is no view component of `Reply`,
just exists as `ListItem`.

All `ListItem` in View **aim to state-less** (no side-effect).
So **every events on `ListItem`** that requires AJAX (like edit, delete, set sticky, etc...) is propagated to `index.js (Container component)`, then processed by container component.

So roles of each `js` likes below:

- `index.js`: handle functions and events having side-effects, wrapping components.
- `ForumComponent`: Show navigation, pagination, list of children(`Forum` and `Topic`), and creation forms.
- `TopicComponent`: Show navigation, pagination, show content of the `Topic`, list of children(`Reply`), and creation forms.
- `items/ForumItem`: Brief information of Forum to be listed in `ForumComponent`
- `items/TopicItem`: Brief information of Forum to be listed in `ForumComponent`
- `items/ReplyItem`: Brief information of Forum to be listed in `TopicComponent`. Include editing-form

Currently, `create***Form` has side-effects. (Not removed yet)
It would be modified before merge this branch to `dev`.

### CODE REVIEW

File: client/src/components/CurrentUserInfo.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const CurrentUserInfo = (props) => {
+    if (props.currentUser) {
+        return (
+            <div>
+                <span>Hi {props.currentUser.name}</span><br/>
+                <button onClick={props.onLogout}>Logout</button>
+            </div>
+        );
+    }else{
```
Comment (jkjh0819) [inline client/src/components/CurrentUserInfo.js]: Could you add a space between if-end bracket and else, else and else-start bracket?

File: client/src/components/Navigation.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const {node} = props;
+
+    let ancestorList;
+    if (node.hasOwnProperty('ancestorList')) {
+        // If parent node does have an ancestor list.
+        // Add itself to the ancestor list.
+        // var parentNodeAncestorList = this.props.parentNode.ancestorList does copy by REFERENCE (original object updated after push).
+        ancestorList = JSON.parse(JSON.stringify(node.ancestorList));
+        ancestorList.push({_id: node._id, title: node.title});
+    }else {
```
Comment (jkjh0819) [inline client/src/components/Navigation.js]: Could you add a space between if-end bracket and else?

File: client/src/containers/NodeView/ForumComponent/createTopicForm.js [hunk 3]
```diff
@@ -1,17 +1,16 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import { Field, reduxForm, SubmissionError } from 'redux-form'
-import { blurIfNoPermission } from '../utils/permissionChecker'
-import { nodeListFetch } from '../actions/nodeListActions'
+import { blurIfNoPermission } from '../../../utils/permissionChecker'
```
Comment (jkjh0819) [inline client/src/containers/NodeView/ForumComponent/createTopicForm.js]: Please choose whether to use semicolon at import or not

**Comment by Prev** [comment]
### 4. Renew UI

Renew UI to `bootstrap4` style, remove redux-form.css.

Some feature (Create forum form) is currently hidden because it will be executed in `admin` page, not client.

**Pull request merged**